### ISSUE TITLE
Openstack instance: metadata is not removed

### ISSUE THREAD

**Issue by michalmedvecky** [hashicorp/terraform#11239]
Terraform does not remove metadata when you remove it from .tf file.

TLDR: 

1. create any new openstack_v2 instance with some_metadata = some_value
2. modify your tf file so that some_metadata does not exist anymore and create another one (some_metadata2);  terraform apply
3. result: your instance now has metadata `some_metadata` and `some_metadata2`

### Terraform Version
Terraform v0.8.4

### Affected Resource(s)
- openstack_compute_instance_v2

### Terraform Configuration Files

create.tf:

```hcl
provider "openstack" {
  alias = "mytenant"
  user_name = "mmedvecky"
  tenant_name = "mytenant"
  auth_url = "http://1.2.3.4:5000/"
}

resource "openstack_compute_instance_v2" "mmedvecky-test01" {
  provider = "openstack.mytenant"
  name = "mmedvecky-test01"
  image_id = "5a0bdb06-5dfe-4202-8f14-0de39156ee43"
  flavor_name = "m1.small"
  key_pair = "misko"
  availability_zone = "mz"
  security_groups = ["default"]

  metadata {
    test_metadata = "test"
  }

  network {
    name = "mytenantnetwork"
  }

}
```

modify.tf:

```hcl
provider "openstack" {
  alias = "mytenant"
  user_name = "mmedvecky"
  tenant_name = "mytenant"
  auth_url = "http://1.2.3.4:5000/"
}

resource "openstack_compute_instance_v2" "mmedvecky-test01" {
  provider = "openstack.mytenant"
  name = "mmedvecky-test01"
  image_id = "5a0bdb06-5dfe-4202-8f14-0de39156ee43"
  flavor_name = "m1.small"
  key_pair = "misko"
  availability_zone = "mz"
  security_groups = ["default"]

  metadata {
    test_metadata2 = "test"
  }

  network {
    name = "mytenantnetwork"
  }

}
```

### Expected Behavior
`test_metadata` should be removed

### Actual Behavior
`test_metadata` still exists

### Steps to Reproduce

1. Copy `create.tf` to your terraform directory
2. Run `terraform apply` - your VM will be created
3. edit create.tf, modify `test_metadata` -> `test_metadata2`
4. run `terraform apply`

```
ubuntu@deployer:~/iam-deployment/tftest$ terraform apply
openstack_compute_instance_v2.mmedvecky-test01: Refreshing state... (ID: 15d89d43-cb2a-4af9-a560-8adbe8b2f387)
openstack_compute_instance_v2.mmedvecky-test01: Modifying...
  metadata.test_metadata:  "test" => ""
  metadata.test_metadata2: "" => "test"
openstack_compute_instance_v2.mmedvecky-test01: Modifications complete

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

State path: terraform.tfstate
```

5. run `terraform apply` again:

```
ubuntu@deployer:~/iam-deployment/tftest$ terraform apply
openstack_compute_instance_v2.mmedvecky-test01: Refreshing state... (ID: 15d89d43-cb2a-4af9-a560-8adbe8b2f387)
openstack_compute_instance_v2.mmedvecky-test01: Modifying...
  metadata.%:             "2" => "1"
  metadata.test_metadata: "test" => ""
openstack_compute_instance_v2.mmedvecky-test01: Modifications complete

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

State path: terraform.tfstate
```

and again:

```
ubuntu@deployer:~/iam-deployment/tftest$ terraform apply
openstack_compute_instance_v2.mmedvecky-test01: Refreshing state... (ID: 15d89d43-cb2a-4af9-a560-8adbe8b2f387)
openstack_compute_instance_v2.mmedvecky-test01: Modifying...
  metadata.%:             "2" => "1"
  metadata.test_metadata: "test" => ""
openstack_compute_instance_v2.mmedvecky-test01: Modifications complete

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

State path: terraform.tfstate
```

See, it always says modifications complete, but the old removed metadata still persists on the host.

**Comment by jtopjian** [comment]
@michalmedvecky Thank you for the report! 

#11252 should fix this problem :)

**Comment by michalmedvecky** [comment]
Works for my test case! Thanks!
